modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/RuntimeMessageHandler.java

```diff
@@ -29,24 +33,64 @@
 
     /**
      * Warning messages don't block server startup.
+     *
+     * @deprecated since 11.3, use {@link #addMessage(Level, String)} instead.
      */
+    @Deprecated
     void addWarning(String message);
 
     /**
      * @return an unmodifiable {@link List} of warning messages
+     * @deprecated since 11.3, use {@link #getMessages(Predicate)} instead
      */
+    @Deprecated
     List<String> getWarnings();
 
     /**
      * Add new error.
      * <p />
      * Error messages block server startup in strict mode.
+     *
+     * @deprecated since 11.3, use {@link #addMessage(Level, String)} instead.
      */
+    @Deprecated
     void addError(String message);
 
     /**
      * @return an unmodifiable {@link List} of error messages
+     * @deprecated since 11.3, use {@link #getMessages(Predicate)} instead
      */
+    @Deprecated
     List<String> getErrors();
 
+    /**
+     * Adds the following message.
+     *
+     * @since 11.3
+     */
+    void addMessage(Level level, String message);
+
+    /**
+     * Returns all messages, filtered by following predicate.
+     * <p>
+     * Predicate can be null, in which case no filtering will be done.
+     *
+     * @since 11.3
+     */
+    List<RuntimeMessage> getRuntimeMessages(Predicate<RuntimeMessage> predicate);
+
+    /**
+     * Returns all messages strings, filtered by following predicate.
+     *
+     * @since 11.3
+     */
+    List<String> getMessages(Predicate<RuntimeMessage> predicate);
```

Comment:
Worth adding a convenient method `#getMessages(Level)` to ease the replacement of `#getErrors` and `#getWarnings` methods? Make sens to me as we have `#addMessage(Level, String)`.

Reply:
Added this api for convenience